tuf/api/metadata.py

```diff
@@ -1384,6 +1465,58 @@ def _is_target_in_pathpattern(targetpath: str, pathpattern: str) -> bool:
 
         return True
 
+    def _calculate_delegation(self, hash_bits_representation: str) -> str:
```

Comment:
this method now seems like it should be part of the new class (HashDelegations.find_delegation() maybe)?

I also think the argument to the function should be `targetpath` : this way the arguments and output make sense to a reader: you feed in the targetpath and get the real delegated role name as output -- The fact that this specific implementation uses a string of ones and zeros as an intermediate value is not important and you only need to move a few lines of code to get this better abstraction.

Then `DelegatedRole.find_delegation()` ends up being quite clean, something like
```
if self.succinct_hash_info:
    return self.succinct_hash_info.find_delegation(target_filepath)
elif self.is_delegated_path(target_filepath):
    return self.name
return None
```

Opinions?

Reply:
The reason I don't pass `targetpath` as an argument is the testing.
I want to test the actual bin calculation math and if I pass the `targetpath` this means that the hash calculation has to be moved inside `_calculate_delegation` and I won't be able to pass the binary representation in a string as do now. 
I think the test inside `test_api.py` won't be clear for the reader if we do that.
Given that this is an internal API I think it's allowed to require a little more specific argument for easier testing.

Reply:
hmm, I think I'll leave this to you and @lukpueh as I think you've been talking about this part of the code.

Personally I'm not fond of making production code harder to read based on unit testing needs, or the basic idea of building a string of ones and zeros just to turn it into an int later ... but I can live with those so I'll leave these for you